app/assets/javascripts/discourse/app/routes/discovery-index.js

```diff
@@ -14,6 +16,11 @@ export default class DiscoveryIndex extends DiscourseRoute {
     if (params) {
       destination += `&${params}`;
     }
+
+    if (this.siteSettings.login_required && !this.currentUser) {
+      destination = "/login-required?_discourse_homepage_rewrite=1";
```

Comment:
Should we be using the `homepageRewriteParam` "constant" defined in `homepage-router-overrides.js`?